### PR TITLE
2034 feature request button for generate static summary with gene coverages

### PR DESCRIPTION
**Pull request by metehaansever**
I had to create a new endpoint in order not to add a new argument to summarize.py. Of course, this is not the best solution, but I am open to review.

### CODE REVIEW

**Comment by metehaansever** [comment]
I create a checkbox button on the same page.
image

image

**Review by meren — COMMENTED** [review]
Here are some comments. I agree, adding another route just for this is not the best way forward. It leads to redundant code, and I think there is a more elegant solution.

File: anvio/bottleroutes.py [hunk 1]
```diff
@@ -163,6 +163,7 @@ def register_routes(self):
         self.route('/upload_project',                          callback=self.upload_project, method='POST')
         self.route('/data/contig/<split_name>',                callback=self.get_sequence_for_split)
         self.route('/summarize/<collection_name>',             callback=self.gen_summary)
+        self.route('/summarize_full/<collection_name>',        callback=self.gen_summary_full)
```
[the remark this answers is not in the document]
Comment (metehaansever) [reply anvio/bottleroutes.py:166]: If we do it this way, we will overwrite the currently used endpoint. For a certain collection_name, I need to run both GET method and POST something for init_gene_coverages information. Instead, we can run all static summaries from a single endpoint with init_gene_coverage.

[the remark this answers is not in the document]
Comment (meren) [reply anvio/bottleroutes.py:166]: UNLESS, you also send the collection name with a POST variable? I don't understand what is a problem here :)

[the remark this answers is not in the document]
Comment (metehaansever) [reply anvio/bottleroutes.py:166]: No, I don't POST anything. The only difference between the two routes is the callback functions. Both are GET methods.

[the remark this answers is not in the document]
Comment (meren) [reply anvio/bottleroutes.py:166]: Yes, I get that, that's why I suggested in my comment a single POST function instead:

```
self.route('/summarize',             callback=self.gen_summary, method='POST')
```

You said "I need to run both GET method and POST something for init_gene_coverages information" and I said you need to do that UNLESS use a POST method :)

File: anvio/bottleroutes.py [hunk 2]
```diff
@@ -1038,6 +1038,11 @@ def gen_summary(self, collection_name):
         # common params. we will set pan/profile specific params a bit later:
         summarizer_args.collection_name = collection_name
         summarizer_args.taxonomic_level = self.interactive.taxonomic_level
+        init_gene_coverages = request.forms.get('init_gene_coverages')
+        
+        if init_gene_coverages:
+            summarizer_args.init_gene_coverages = True
+            self.interactive.mode == 'full'
```
Comment (meren) [inline anvio/bottleroutes.py:1045]: Best practice would suggest that this is *not* a good place to do it. You are changing the `self.interactive.mode` to `full` if `init_gene_coverages` is present in POST variables.

There is an `elif self.interactive.mode == 'full'` below, and the check for `init_gene_coverages` should be handled under there. I.e.,

``` python
        (...)
        elif self.interactive.mode == 'full':
            summarizer_args.profile_db = self.interactive.profile_db_path
            summarizer_args.contigs_db = self.interactive.contigs_db_path

            if init_gene_coverages:
                summarizer_args.init_gene_coverages = True

            summarizer_args.output_dir = os.path.join(os.path.dirname(summarizer_args.profile_db), 'SUMMARY_%s' % collection_name)
        else:
            (...)
```

**Comment by meren** [comment]
Thanks for these changes, Mete. It looks much better now.

**Comment by meren** [comment]
This is now in the main branch! Thank you, Mete!

**Comment by metehaansever** [comment]
Thank you for all the Review :)

**Pull request merged**